cmd/envcheckctl/README.md

```diff
@@ -0,0 +1,173 @@
+# envcheckctl
+
+The application `envcheckctl` is used to inspect a Kubernetes/OpenShift cluster for common agent
+installation issues.
+
+## Pull Debug Data
+
+The application `envcheckctl` is capable of collecting data directly from the cluster.
+
+```bash
+# use a specific kubeconfig file
+$ envcheckctl inspect -kubeconfig $KUBECONFIG
+
+# use the default context for kubectl
+$ envcheckctl inspect
+```
+As a result of the data pull, the application generates a JSON file with the name like `cluster-info-${TIMESTAMP}.json`.
+The application also populates standard output with the following information:
+
+```yaml
+# Summary information of all resources found
+pods=256, running=256, nodes=19, containers=338, namespaces=13, deployments=56, replicaSets=56, daemonsets=9, statefulsets=7, duration=1.435853996s
+
+# Coverage indicates how many hosts in the cluster is actively monitored by Instana. Generally we expect this to be 100% however it is common to have less than 100% with OpenShift and self-managed Kubernetes clusters whereby the control-plane is not monitored due to taints. Less than 100% coverage in the absence of a taint can be an indicator for broken traces and missing infrastructure metrics.
+coverage
+- "13 of 19 (68.42%)"
+
+# distribution type kubernetes / openshift / eks / gke / aks
+serverDistribution
+ - eks
+
+# Server version is useful for indicating API compatibility and can be used to quickly identify any release related compatibility issues such as deprecations.
+serverVersion
+ - v1.23.14-eks-ffeb93d
+
+# Top 10 agents ordered by number of restarts. This is useful for diagnosing whether any agents in the cluster are experiencing frequent restarts.
+agentRestarts
+- "instana-agent-4j8zq"=5
+- "instana-agent-jt84d"=5
+- "instana-agent-t5tcs"=4
+- "instana-agent-fvgbj"=2
+- "instana-agent-jvbhr"=1
+
+# Agent status indicates the current state of all Instana agents in the cluster. If any agents are not Running then it indicative of potentially anomalous infrastructure metric and trace behaviour.
+agentStatus
+- "Running"=13
+
+# Lists the distribution of helm chart versions and number of pods
+chartVersions
+- "1.2.45"=13
+
+# Distribution of cniPlugin
+cniPlugins
+- "cilium"=19
+
+# Distribution of container Runtimes
+containerRuntimes
+- "containerd://1.6.6"=19
+
+# Distribution of nodes types
+instanceTypes
+- "c5.12xlarge"=4
+- "m5.4xlarge"=9
+- "m5.large"=3
+- "m5.xlarge"=3
+
+# Distribution of kernels on nodes
+kernels
+- "5.4.231-137.341.amzn2.x86_64"=19
+
+# Distribution of kubelets on nodes
```

Comment:
Kubelet lets us know that the cluster is aligned and consistent in it's release.